pdfPreviewInMsg/background.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * This example extension is based on work from:
+ *  - e-gaulue
+ *  - rholeczy
+ */
+
+const PREVIEW_WIDTH = 160;
+const PREVIEW_HEIGHT = PREVIEW_WIDTH * 1.5;
+
+
+/**
+ * Generate a data image URL of a scaled down page of a PDF document.
+ *
+ * @param {PDFDocumentProxy} pdf - the PDF document received from pdf.js
+ * @param {Integer} pageNumber - the page number of the PDF document to be rendered
+ *
+ * returns {String} A data URL of the rendered PDF preview
+ */
+async function generatePdfImageUrl(pdf, pageNumber) {
+  // The PDF will be rendered into a temporary HTML canvas element.
+  let canvas = document.createElement("canvas");
+  canvas.width = PREVIEW_WIDTH;
+  canvas.height = PREVIEW_HEIGHT;
+
+  let page = await pdf.getPage(pageNumber);
+  let viewport = await page.getViewport({ scale: 1.0 });
+  let scale = Math.min(
+    canvas.width / viewport.width,
+    canvas.height / viewport.height
+  );
+  let thumbViewPort = await page.getViewport({ scale });
+  let context = canvas.getContext("2d");
+  await page.render({ canvasContext: context, viewport: thumbViewPort })
+    .promise;
+
+  // Extract the image data URL of the rendered PDF from the canvas.
+  return canvas.toDataURL("image/jpeg");
+}
+
+/**
+ * This function reads all attachments and extracts data image URLs for PDFs and
+ * standard images and sends those to the requesting tab for display.
+ *
+ * The pdf.js library is used to render PDFs.
+ *
+ * @param {Integer} tabId - the id of the WebExtension message display Tab
+ * @param {Integer} messageId - the id of the WebExtension MessageHeader
+ */
+async function addInlinePreviews(tabId, messageId) {
+  // Get a list of all attachments.
+  let attachments = await browser.messages.listAttachments(messageId);
+  let loadingSpinnerShown = false;
+
+  for (let attachment of attachments) {
+    // Only image or a PDF attachments are handled.
+    if (
+      !attachment.contentType.toLowerCase().startsWith("application/pdf") &&
+      !attachment.contentType.toLowerCase().startsWith("image/")
+    ) {
+      continue;
+    }
```

Comment:
Early exit if the condition is not meet improves readability.

Reply:
Sorry but when we put a "!" in front of a boolean var, it's don't mean :
```
if ( var == false) {}
```
?

Reply:
It's just for the readability ok , i didn't the "continue" in a for, thanks